### ISSUE TITLE
Feature req: Show int/uint buffer values as hex in buffer viewers

### ISSUE THREAD

**Issue by hrydgard** [baldurk/renderdoc#2708]
## Description

Sometimes uint values are used not just for numbers but also for things like bitmasks in shaders. Would be very useful to have the option to display them in hex like you can in VS' watch windows. This is a bit hard to parse without copy-pasting to calc.exe:

![image](https://user-images.githubusercontent.com/130929/187309986-f5a04084-089b-4194-a065-f58949cb6d0a.png)

Specifically, I think a global toggle for display of int/uint variable as hex, togglable from the right click menu in the buffer viewer similar to in VS, would do just fine. 

## Environment

* RenderDoc version: 1.21
* Operating System: Windows 10
* Graphics API: Vulkan

**Comment by baldurk** [comment]
If you add `[[hex]]` to the buffer format specification then integers will be displayed as hex. That sounds like it would do what you're asking for?

**Comment by hrydgard** [comment]
This is without a buffer format specification, just looking at the contents of a uniform buffer.

Although yeah, that does work if I apply one! That is enough for me right now, although perhaps not maximally convenient. Feel free to close.

**Comment by baldurk** [comment]
I think this is how I would like to solve it for now then, I'm not sure that having a global hex/dec toggle is the right choice. I'll bear this in mind in case this comes up again. What would be nice is some way of attaching shader-based metadata to indicate that the variable should be interpreted as hex/bin/oct or whatever, but that's overengineering for just this.
